core/core-uri/src/jbMain/kotlin/androidx/core/uri/Uri.jb.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.core.uri
+
+public actual abstract class Uri {
+    actual abstract fun getFragment(): String?
+    actual abstract fun getQuery(): String?
+    actual abstract fun getPathSegments(): List<String>
+    actual open fun getQueryParameters(key: String): List<String> = error("Abstract implementation")
+    actual open fun getQueryParameterNames(): Set<String> = error("Abstract implementation")
+    actual abstract override fun toString(): String
+}
+
+public actual object UriUtils {
+    actual fun encode(s: String?, allow: String?): String? = InternalUri.encode(s, allow)
+    actual fun decode(s: String?): String? = InternalUri.decode(s)
+    actual fun parse(uriString: String?): Uri = ActualUri(uriString!!)
```

Comment:
Can we get rid of `!!` here?

Reply:
This is method convention. It throws an exception if the input is null

Reply:
refactored